test/utils.py

```diff
@@ -0,0 +1,51 @@
+from collections.abc import Sequence
+from dataclasses import dataclass
+
+import numpy as np
+from gymnasium import spaces
+
+from tianshou.data.collector import CollectStats
+
+
+@dataclass
+class SpaceInfo:
+    action_shape: int | Sequence[int]
+    state_shape: int | Sequence[int]
+    action_dim: int
+    state_dim: int
+    min_action: float
+    max_action: float
+
+
+def get_space_info(
+    space: spaces.Space,
+) -> SpaceInfo:
+    if isinstance(space, spaces.Box):
+        return SpaceInfo(
+            action_shape=space.shape,
+            state_shape=space.shape,
+            action_dim=space.shape[0],
+            state_dim=space.shape[0],
+            min_action=float(np.min(space.low)),
+            max_action=float(np.max(space.high)),
+        )
+    elif isinstance(space, spaces.Discrete):
+        return SpaceInfo(
+            action_shape=int(space.n),
+            state_shape=int(space.n),
+            action_dim=int(space.n),
+            state_dim=int(space.n),
+            min_action=float(space.start),
+            max_action=float(space.start + space.n - 1),
+        )
+    else:
+        raise NotImplementedError("Unsupported space type")
```

Comment:
Log messages should be as informative as possible, otherwise it's difficult to make use of them.

Pls change to

```python
raise NotImplementedError(f"Unsupported space type: {space.__class__}. Currently supported types are Discrete and Box")
```

Reply:
I also think it's more of a `ValueError`

Reply:
Fixed.